x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/FallbackClaim.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+
+import java.text.ParseException;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.core.Strings.format;
+
+public class FallbackClaim {
+    private final String name;
+    private final JWTClaimsSet claimsSet;
+    private final String actualName;
+
+    public FallbackClaim(String name, Map<String, String> fallbackClaimNames, JWTClaimsSet claimsSet) {
+        this.name = name;
+        final String fallbackName;
+        if (fallbackClaimNames != null) {
+            fallbackName = fallbackClaimNames.getOrDefault(name, name);
+        } else {
+            fallbackName = null;
+        }
+        if (fallbackName == null) {
+            this.actualName = name;
+        } else {
+            this.actualName = claimsSet.getClaim(name) != null ? name : fallbackName;
+        }
+        this.claimsSet = claimsSet;
+    }
+
+    public String getActualName() {
+        return actualName;
+    }
+
+    public String getStringClaimValue() {
+        try {
+            return claimsSet.getStringClaim(actualName);
+        } catch (ParseException e) {
+            throw new IllegalArgumentException(format("cannot parse string claim [%s] as string", this), e);
+        }
+    }
+
+    public List<String> getStringListClaimValue() {
```

Comment:
nit: (purely style) getClaimValue  and  getClaimValues  instead of getStringClaimValue and getStringListClaimValue ? (the return types are evident and shorter method names without loosing info helps with readability )...and I guess that this also extends to the class `JwtStringClaimValidator` to call it just `JwtClaimValidator`  (where string is implied as the default)

Reply:
Patterns of these method names follow the corresponding method names defined in `JWTClaimsSet`, which has `getStringClaim` and `getStringListClaim`. This is because there are other types of data that can be retreived from a JWT, e.g. `getDateClaim`. Since return type is *not* part of the Java method signature, the difference has to be explicitly embedded in the method names.

The `FallbackableClaim` does not currently support other data type. So it is possible to drop `String` and `StringList` from the method names. But the class itself does not prevent future expansion to include other non-string data types. So I'd keep it as is which is also consistent with the namings from JWTClaimsSet. 

Similarly, there is also a `JwtDateClaimValidator` in addition to `JwtStringClaimValidator`. So I'll keep the type name in the class name. We may be able to drop them by using generics. But I don't think it is worth the effort.